tgui/packages/tgui/interfaces/MapManipulation.js

```diff
@@ -0,0 +1,1203 @@
+                });
+              }}
+            />
+          </Stack.Item>
+          <Stack.Item>
+            <Button
+              content="Flicker Lights SW"
+              tooltip="Flickers some lightbulbs in that maintenance area for a short, random interval."
+              onClick={() => {
+                act('flicker_light', {
+                  area_to_flicker:
+                    '/area/whiskey_outpost/blackstone/underground/maintenance/south_west',
+                });
+              }}
+            />
+          </Stack.Item>
+        </Stack>
+      </Section>
+    </Stack>
+  );
+};
+
+/*
+Cannot use local state to track the values here as someone else can modify them
+at any time; as such, this menu will lag a little more than the others, but
+that should be okay. Suppose it is possible to use local state, but that would
+mean changes made by other people would be hidden from view. Perhaps something
+to rework later.
+*/
+
+import { map } from 'common/collections';
+
+export const NightmareTab = (props, context) => {
+  const { data, act } = useBackend(context);
+  const [sel_nigh_map, setNightMap] = useLocalState(
+    context,
+    'sel_nigh_map',
+    'ground_map'
+  );
+
+  return (
+    <Stack vertical>
+      <Section
+        title="Map Selection"
+        buttons={
+          <Button
+            color="purple"
+            content="Prepare Game"
+            tooltip="Fires the Nightmare system early, setting up all of the components before round start."
+            onClick={() => {
+              act('nightmare_prepare_game');
+            }}
+          />
+        }>
+        <Stack textAlign="center">
+          <Stack.Item grow={1}>
+            <Button
+              fluid
+              icon="globe"
+              selected={sel_nigh_map === 'ground_map'}
+              content="Ground Map"
+              tooltip="Looks up ground nightmare values."
+              onClick={() => {
+                setNightMap('ground_map');
+              }}
+            />
+          </Stack.Item>
+          <Stack.Item grow={1}>
+            <Button
+              fluid
+              icon="rocket"
+              selected={sel_nigh_map === 'ship_map'}
+              content="Ship Map"
+              tooltip="Looks up ship nightmare values."
+              onClick={() => {
+                setNightMap('ship_map');
+              }}
+            />
+          </Stack.Item>
+        </Stack>
+      </Section>
+      {sel_nigh_map === 'ground_map' ? (
+        <NightmareGroundPanel />
+      ) : (
+        <NightmareShipPanel />
+      )}
+    </Stack>
+  );
+};
+
+// These two menus can be one menu, but some other time.
+
+export const NightmareGroundPanel = (props, context) => {
+  const { data, act } = useBackend(context);
+  const NightmareGround = data.nightmare_ground || {};
+  const NightmareGroundScenario = data.nightmare_ground_scenario || [];
```

Comment:
uppercase in jsx is for components (hence why these variables are syntax highlighted orange here)

```suggestion
  const nightmareGround = data.nightmare_ground || {};
  const nightmareGroundScenario = data.nightmare_ground_scenario || [];
```